subsys/net/l2/openthread/openthread.c

```diff
@@ -227,17 +240,74 @@ void ot_receive_handler(otMessage *aMessage, void *context)
 	otMessageFree(aMessage);
 }
 
+static struct openthread_context *ot_context_for_restarter;
+
+void ot_joiner_start_handler(otError error, void *context);
+
+static void ot_joiner_restarter(void *dummy)
+{
+	otError error = otJoinerStart(ot_context_for_restarter->instance,
+					joiner_pskd, NULL,
+					PACKAGE_NAME, OT_PLATFORM_INFO,
+					PACKAGE_VERSION, NULL,
+					&ot_joiner_start_handler,
+					ot_context_for_restarter);
+
+	if (error != OT_ERROR_NONE) {
+		NET_ERR("Failed to start joiner [%d]", error);
+	}
+}
+
+K_WORK_DEFINE(joiner_restart, ot_joiner_restarter);
+
+static void ot_joiner_restart_timer(void *dummy)
+{
+	k_work_submit(&joiner_restart);
+}
+
+K_TIMER_DEFINE(joiner_restart_timer, ot_joiner_restart_timer, NULL);
 void ot_joiner_start_handler(otError error, void *context)
 {
 	struct openthread_context *ot_context = context;
-
+	static u32_t restart_delay;
```

Comment:
Also, looking at PR #25679 which got just merged,it seems we should use `uint32_t` now.